### ISSUE TITLE
cmd/compile: corruption on linux-amd64-noopt builder

### ISSUE THREAD

**Issue by rsc** [golang/go#55156]
```
#!watchflakes
post <- builder == "linux-amd64-noopt" && date < "2023-06-01"
```

The noopt builder is clearly having some kind of corruption problem (see #54779, #54906, #54907). 
Those will be closed as duplicates of this one, once automated flake triage is working.

**Comment by randall77** [comment]
I've been looking at these issues. My current theory is that pointer bitmaps for stack frames are somehow wrong.
I came across an instance which reported that a stack slot which should contain a pointer, contains something obviously not a pointer. But looking at the generated assembly, that stack slot should not even be live at that point (it is both initialized and used only after the call at which the function was suspended).

Fairly reliable reproducer:
```
GOGC=1 GO_GCFLAGS='-N -l' ./make.bash
```
Usually it is the compiler that crashes while building the stdlib as the last step of make.bash.

The frame with the bad pointer always seems to be suspended at a call to `runtime.typedmemclr`.

**Comment by randall77** [comment]
`typedmemclr` should not be suspendible. Something it calls is suspending when it shouldn't.

Seems to be fixed by adding a `//go:nosplit` to `(*mSpanStateBox).get`.

Looks like a good use for `nosplitrec`.

**Comment by randall77** [comment]
Maybe caused by CL 424395? Not that it is that CL's fault, but that CL introduces a (non-intrinsic) call to the body of `(*mSpanStateBox).get`, which causes `get` to have a prologue, and thus a suspension point.

**Comment by randall77** [comment]
Fix is in. I optimistically assumed that the other issues referenced above are also fixed. If not, we'll soon find out...

**Comment by rsc** [comment]
Thanks for the fix! Still using this as a test case for the automated issue triage, so feel free to ignore the following message from gopherbot.

**Comment by rsc** [comment]
These are not newer than Keith's fix. Closing. Bot will reopen if we find any new ones.

**Comment by randall77** [comment]
That last flake doesn't make any sense. Particularly, that it is flaky. The noinit.go test it is flaking on is completely deterministic.
Also it doesn't really have anything to do with the other failures in this issue, except for the builder it happened on.
I'm going to reclose. I guess if we see more of these I'll make a new issue for it.

**Comment by bcmills** [comment]
These are unrelated failures. Added a `date` constraint to the pattern, and closing. `watchflakes` can file new issues for new failure modes.
